python_modules/libraries/dagster-components/dagster_components/utils.py

```diff
@@ -74,9 +74,33 @@ def get_path_for_package(package_name: str) -> str:
 @dataclass
 class TranslatorResolvingInfo:
     obj_name: str
-    asset_attributes: AssetAttributesModel
+    asset_attributes: Union[str, AssetAttributesModel]
     resolution_context: ResolutionContext
 
+    def _resolve_asset_attributes(
+        self, context: Mapping[str, Any]
+    ) -> Union[AssetSpec, AssetAttributesModel]:
+        """Resolves the user-specified asset attributes into an AssetAttributesModel, or an AssetSpec
+        if the UDF returns one.
+        """
+        if isinstance(self.asset_attributes, AssetAttributesModel):
+            return self.asset_attributes
+
+        resolved_asset_attributes = self.resolution_context.with_scope(**context).resolve_value(
+            self.asset_attributes
+        )
+
+        if isinstance(resolved_asset_attributes, AssetSpec):
+            return resolved_asset_attributes
+        elif isinstance(resolved_asset_attributes, AssetAttributesModel):
+            return resolved_asset_attributes
```

Comment:
can you get `AssetAttributesModel` post resolve?

Reply:
if the user fn returns an `AssetAttributesModel` explicitly